scalafix-rules/src/main/scala/scalafix/internal/rule/CompilerTypePrinter.scala

```diff
@@ -107,9 +107,22 @@ class CompilerTypePrinter(g: ScalafixGlobal, config: ExplicitResultTypesConfig)(
         extraPatch += patch
         tpe
     }
-    val preProcessed = preProcess(toLoop, inverseSemanticdbSymbol).widen
-    val shortType = g.shortType(preProcessed, history)
-    val short = shortType.toString()
+
+    def typeToString(typ: Type): String = {
+      val preProcessed = preProcess(typ, inverseSemanticdbSymbol).widen
+      val shortType = g.shortType(preProcessed, history)
+      val shortTypeSymbol = shortType.typeSymbol
+
+      shortTypeSymbol.fullName match {
+        case "scala.Tuple1" => // toString of Tuple1[Foo] is (Foo,) which is invalid syntax
```

Comment:
I guess so. I'm not brave enough to try fixing it upstream though

Reply:
Looking into this further, there are a *lot* of cases where `toString` from scala reflection doesn't produce correct code for the reflected type (e.g. no insertion of backticks). I think the only way to have really robust generation of type annotations will be to completely reimplement it.

Reply:
We don't use a toString to print the type, but a custom recursive pprint (for pretty print).  I agree with you on this, this need to be rewritten and simplified: there are many bugs that has been fixed without a rewrite, which makes the code more complicated, with too many special cases. 
We can move forward and fix this one, but in fact a rewrite is already needed for the `tostring or prettyprint`

Reply:
> We don't use a toString to print the type, but a custom recursive pprint (for pretty print).

Can you elaborate on this? The `Type` type here is from scala-reflect, not scalameta, so it does seem to be using the scala-reflect `toString` here. Do we need to convert it to a scalameta type in order to prettyprint it?

Reply:
I understand what you mean. In fact we do a toString at the end, but before we build the Type in different methods like 
```
  private def preProcess(tpe: Type, gsym: Symbol): Type = { 
  ....
        case tp @ ThisType(sym)
            if tp.toString() == s"${gsym.owner.nameString}.this.type" =>
          new PrettyType("this.type")
          
```
The class PrettyType is a scalafix Type that  extends Type from the compiler. So for example if we have a `ThisType`, we will print `this.type`. 
So what I wanted to say, is that all the work is done in 2 or 3 methods, that write how the type is supposed to be printed (`def preProcess and def shortType`)


I don't know if the pretty print of scalameta always produces string (ResultType) that compiles, but in all cases, we don't use scalameta for two reasons:
- semanticdb files don't always have the information about result type, so we rely on the presentation compiler to have the typed tree.
- I don't think that it's easy to shorten the result type based only on scalameta information (but maybe I m wrong on this. The only thing I know that this approach is now used in metals and here, but maybe we need to questing it one day)

Reply:
I see, thanks for the clarification. Does that mean my fix should be done in the toStrings of the pretty types rather than where it currently is?

Reply:
In fact, ideally your fix should be inside shortType, somewhere inside this huge function
like this
```
  def shortType(longType: Type, history: ShortenedNames): Type = {
    val isVisited = mutable.Set.empty[(Type, Option[ShortName])]
    val cached = new ju.HashMap[(Type, Option[ShortName]), Type]()
    def loop(tpe: Type, name: Option[ShortName]): Type = {
      val key = tpe -> name
      // NOTE(olafur) Prevent infinite recursion, see https://github.com/scalameta/metals/issues/749
      if (isVisited(key)) return cached.getOrDefault(key, tpe)
      isVisited += key
      val result = tpe match {
        case TypeRef(pre, sym, args) =>
          if (history.isSymbolInScope(sym, pre)) {
            sym.name match {
              case TypeName("Tuple1") =>
              // it means the args is a list of one element
                new PrettyType(s"Tuple1[${loop(args.head, None)}]")
              case _ => TypeRef(NoPrefix, sym, args.map(arg => loop(arg, None)))
            }
            ````

Reply:
for example your code wouldn't work with
def foo = List(Tuple1(5))

I m not sure about completly on how to add this in this huge method, but it will work with nested types 
``.
            sym.name match {
              case TypeName("Tuple1") =>
              // it means the args is a list of one element
                new PrettyType(s"Tuple1[${loop(args.head, None)}]")
              case _ => TypeRef(NoPrefix, sym, args.map(arg => loop(arg, None)))
            }
```

Reply:
I think the new changes do the job.